client/src/main/java/io/pravega/client/segment/impl/SegmentInputStream.java

```diff
@@ -84,10 +72,11 @@ public default ByteBuffer read() throws EndOfSegmentException, SegmentTruncatedE
     public abstract void close();
     
     /**
-     * Returns true if {@link #read()} can be invoked without blocking. (This may be because there
-     * is data in a buffer, or the call will throw EndOfSegmentException).
+     * Returns > 0 if {@link #read()} can be invoked without blocking. 
+     * Returns 0 if {@link #read()} will block. 
+     * Returns -1 if a call to read will throw EndOfSegmentException.
      *
-     * @return False if data read is blocking.
+     * @return 0 if data read is blocking.
      */
-    public boolean isSegmentReady();
+    public int bytesInBuffer();
```

Comment:
I wanted to confirm that these API changes are not going to break anything. The methods here are public, so in principle, this could be problematic, but I don't think we tell anyone outside Pravega to use this directly.

Reply:
It is not meant to change anything. This is in the impl package and is not considered part of our public API. It needs to be declared public because it needs to be invoked from another Java package, but not anything outside of the client.